### PR TITLE
Sed 3125 rerun isolated execution of ap

### PR DESCRIPTION
**Pull request by iegorov777**

### CODE REVIEW

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java [hunk 1]
```diff
@@ -70,67 +103,168 @@ public TestSetStatusOverview getTestSetStatusOverview(Map<String, String> reposi
...
+            // Here we resolve the original AP file used for previous isolated execution and re-use it to create the execution context
+            Resource resource = getResource(contextId, apName);
+
+            File apFile = null;
+
+            ResourceRevisionFileHandle fileHandle = resourceManager.getResourceFile(resource.getId().toString());
+            if (fileHandle != null) {
+                apFile = fileHandle.getResourceFile();
+            }
+            updateLastExecution(resource);
```
Comment (david-stephan) [inline step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java:122]: do we really want to update the last execution if the ap file was not found

**Review by david-stephan — DISMISSED** [review]
Approved with minor comments

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java (hunk 1 above)
Comment (iegorov777) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java:122]: @david-stephan 
Now I've put this update after the check on `apFile==null`

File: step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepositoryPlugin.java [hunk 2]
```diff
@@ -18,21 +18,125 @@
...
         context.put(IsolatedAutomationPackageRepository.class, repository);
+
+        AutomationPackageExecutor packageExecutor = new AutomationPackageExecutor(
+                context.getScheduler(),
+                context.require(ExecutionAccessor.class),
+                repository
+        );
+        context.put(AutomationPackageExecutor.class, packageExecutor);
+
+        // cron is defined here but not within 'initializeData' step, because we need the default value to be defined before we create the HousekeepingJobsManager
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepositoryPlugin.java:77]: @david-stephan 
All this pre-init logic is now moved to `initializeData`.

**Review by david-stephan — APPROVED** [review]

**Pull request merged**